### PR TITLE
Added deleteStaleUser, getUserDetails and createUserWithSSOId methods…

### PR DESCRIPTION
**Pull request by ashwini-mv**
### JIRA link ###
https://tools.hmcts.net/jira/browse/SIDM-7071

### Change description ###
Added deleteStaleUser, getUserDetails and createUserWithSSOId methods to apiHelper
Added scenarios to test existing mange user with active, stale and suspended status
Added tests to check sso-Id, userId, createDate and Last-modified
Created random stale-user and suspended-user per test
Small html change to add id to create date and last modified

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/config.ts [hunk 1]
```diff
@@ -46,18 +37,18 @@ export const testAccounts = {
     email: 'superuser@test.com',
     password: 'Pa55word11',
```
Comment (KianKwa) [inline src/test/config.ts:38]: You have a constant called PASSWORD at the top of this class, can you not use 'config.PASSWORD' here for all the users' password.

File: src/test/config.ts [hunk 2]
```diff
@@ -46,18 +37,18 @@ export const testAccounts = {
...
     role: 'IDAM_SUPER_USER',
-    firstName: 'superTest'
+    firstName: 'superTest',
   },
   adminUser: {
     email: 'adminuser@test.com',
     password: 'Pa55word11',
     role: 'IDAM_ADMIN_USER',
-    firstName: 'adminTest'
+    firstName: 'adminTest',
```
Comment (KianKwa) [inline src/test/config.ts:46]: I wonder what is the benefit of adding a trailing comma here? In eslint.rc we set the lint rule to be "comma-dangle": ["error", "only-multiline"]...so it does not matter whether there is a trailing comma/

File: src/test/functional/shared/apiHelpers.ts [hunk 3]
```diff
@@ -18,12 +19,34 @@ export const createUserWithRoles = async (email, password, forename, userRoles)
       {
         headers: {'Content-Type': 'application/json'},
       })).data;
-
   } catch (e) {
     throw new Error(`Failed to create user ${email} with roles ${userRoles}, http-status: ${e.response?.status}`);
   }
 };
 
+export const createUserWithSsoId = async (email, password, forename, userRoles,ssoId) => {
```
Comment (KianKwa) [inline src/test/functional/shared/apiHelpers.ts:27]: Missing white space between userRoles and ssoId

Comment (KianKwa) [inline src/test/functional/shared/apiHelpers.ts:27]: This method here is virtually identical to the one before with an additional ssoId field. We can use the same method and check for whether ssoId is undefined, and return the additional field for ssoId in the json of that it present. Read stack overflow here for how to do it https://stackoverflow.com/questions/456177/function-overloading-in-javascript-best-practices :)

File: src/test/functional/shared/apiHelpers.ts [hunk 4]
```diff
@@ -74,16 +110,16 @@ export const getOIDCToken = async () => {
   }
 };
 
-export const suspendUser = async (userId) => {
+export const suspendUser = async (userId,email) => {
```
Comment (KianKwa) [inline src/test/functional/shared/apiHelpers.ts:113]: missing white space between the 2 parameters

File: src/test/functional/shared/apiHelpers.ts [hunk 5]
```diff
@@ -94,6 +130,24 @@ export const suspendUser = async (userId) => {
   }
 };
 
+export const convertISODateTimeToUTCFormat = (date: string): string => {
```
Comment (KianKwa) [inline src/test/functional/shared/apiHelpers.ts:133]: Is there no way you can call the actual method in the "section" section? This looks identical to that

File: src/test/functional/manage-existing-user-test.ts [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+  const email = await I.grabTextFrom('#email');
+  Assert.equal(email.trim(), staleUserEmail);
+
+  const status = await I.grabTextFrom('#status');
+  Assert.equal(status.trim(), 'Archived');
+
+  await deleteStaleUser(user.id);
+});
+
+
```
Comment (KianKwa) [inline src/test/functional/manage-existing-user-test.ts:118]: This test class here looks like superset of the tests in manage-users-test.ts....may be we don't need the other test now? If so you can rename this class to simply manage-users-test.ts

File: src/test/functional/shared/apiHelpers.ts (hunk 5 above)
Comment (ashwini-mv) [reply src/test/functional/shared/apiHelpers.ts:133]: All methods in test suite should be independent of main section, if possible

File: src/test/functional/manage-existing-user-test.ts (hunk 6 above)
Comment (ashwini-mv) [reply src/test/functional/manage-existing-user-test.ts:118]: Fixed

File: src/test/functional/shared/apiHelpers.ts (hunk 4 above)
Comment (ashwini-mv) [reply src/test/functional/shared/apiHelpers.ts:113]: Fixed

File: src/test/functional/shared/apiHelpers.ts (hunk 3 above)
Comment (ashwini-mv) [reply src/test/functional/shared/apiHelpers.ts:27]: Fixed

File: src/test/config.ts (hunk 2 above)
Comment (ashwini-mv) [reply src/test/config.ts:46]: Fixed

File: src/test/config.ts (hunk 1 above)
Comment (ashwini-mv) [reply src/test/config.ts:38]: Removed extra constants

File: src/test/functional/shared/apiHelpers.ts (hunk 5 above)
Comment (ashwini-mv) [reply src/test/functional/shared/apiHelpers.ts:133]: Imported method from main

**Review by KianKwa — APPROVED** [review]

**Pull request merged**